Pyrado/pyrado/algorithms/meta/sprl.py

```diff
@@ -0,0 +1,573 @@
+            cov_chol_flat = to.tensor(cov_chol_flat).double()
+
+        return ParameterAgnosticMultivariateNormalWrapper(
+            mean=mean,
+            cov_chol_flat=cov_chol_flat,
+            mean_is_parameter=self._mean_is_parameter,
+            cov_is_parameter=self._cov_is_parameter,
+        )
+
+    def parameters(self) -> List[to.Tensor]:
+        """ Gets the list of parameters according to the values passed to the constructor. """
+        params = []
+        if self._mean_is_parameter:
+            params.append(self.mean)
+        if self._cov_is_parameter:
+            params.append(self.cov_chol_flat)
+        return params
+
+    def get_stacked(
+        self, return_mean_cov_indices: bool = False
+    ) -> Union[np.ndarray, Tuple[np.ndarray, Optional[List[int]], Optional[List[int]]]]:
+        """
+        Like `MultivariateNormalWrapper.get_stacked(self)`, returns a numpy array with the mean and standard deviations
+        stacked on top of each other if the respective value is a parameter. Additionally, if `return_mean_cov_indices`
+        is `True`, also the indices of the mean/cov values are returned which can be used for addressing them.
+
+        :param return_mean_cov_indices: if `True`, additionally to just the parameters also the indices are returned
+        :return: if `return_mean_cov_indices` is `True`, a tuple `(stacked, mean_indices, cov_indices)`, where the
+                 latter two might be `None` if the respective value is not a parameter; if `return_mean_cov_indices`
+                 is `False`, just the stacked values; the stacked values can have shape `(0,)`, `(k,)`, or `(2 * k,)`,
+                 depending on if none, only mean/cov, or both are parameters, respectively
+        """
+        parameters = self.parameters()
+        stacked = np.concatenate([p.detach().numpy() for p in parameters])
+
+        if return_mean_cov_indices:
+            pointer = 0
+            mean_indices, cov_indices = None, None
+            if self._mean_is_parameter:
+                mean_indices = list(range(pointer, pointer + self.dim))
+                pointer += self.dim
+            if self._cov_is_parameter:
+                cov_indices = list(range(pointer, pointer + self.dim))
+                pointer += self.dim
+            return stacked, mean_indices, cov_indices
+
+        return stacked
+
+
+class SPRL(Algorithm):
+    """
+    Self-Paced Reinforcement Leaner (SPRL)
+
+    This algorithm wraps another algorithm. The main purpose is to apply self-paced RL [1].
+
+    .. seealso::
+        [1] P. Klink, H. Abdulsamad, B. Belousov, C. D'Eramo, J. Peters, and J. Pajarinen,
+        "A Probabilistic Interpretation of Self-Paced Learning with Applications to Reinforcement Learning", arXiv, 2021
+    """
+
+    name: str = "sprl"
+
+    def __init__(
+        self,
+        env: DomainRandWrapper,
+        subroutine: Algorithm,
+        kl_constraints_ub,
```

Comment:
what type does kl_constrains_ub have?

Reply:
Fixed in 4390e453aebce7c38b93d9ba7776f1cc5b9546b0